Documentation/building/dev-workflow.md

```diff
@@ -0,0 +1,46 @@
+Dev Workflow
+===============
+
+The dev workflow describes the development process to follow when working in our repo. 
+It is divided into specific tasks that are fast, transparent and easy to understand.
+
+This provides flexibility as with discrete tasks people can iterate on a specific one without needing to run the full workflow, other processes can reuse the tasks as needed and anyone can put together the tasks as wanted to create a customized workflow.
+
+## Process 
+![Dev Workflow process](../images/dev-workflow.png)
+
+All the tasks (except enlist, code, submit) will have scripts located at the root of the repo, and will have usage help to better understand what they provide.
+In addition, we want to have an “easy button” script for running many or all of these tasks sequentially in a single command.
+
+## Tasks
+
+**Clean**
+
+Cleans the local enlistment/environment, for example removing files not in the repository. 
+
+**Sync**
+
+The sync task gets the latest source history and packages in order to eliminate all the network traffic when we build. This way we are able to hit the network only when we are intentional about doing it and then be able to build even in offline mode. 
```

Comment:
We need to break apart the steps but with an "easy mode" button devs won't have to worry too much about it, but they will have flexibility to run the step they want according to their needs.

Also, the current DW is not covering working with individual projects, so for now you'll do it as you are used to, but keep in mind we are going to and as @weshaggard  said it is our goal to improve the productivity